src/app/components/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { ReplaySubject, takeUntil } from 'rxjs';
+import { OneProductService } from '../../api/oneproduct.service';
+import { IOneSingleProduct } from '../../Interface/singleproduct';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss'],
+})
+export class ProductDetailComponent implements OnInit, OnDestroy {
+  oneProduct!: IOneSingleProduct | undefined;
+
+  destroy$ = new ReplaySubject<void>(1);
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    public oneProductService: OneProductService
+  ) {}
+
+  openConfig(param: string): void {
+    this.router.navigate(['configurator', param]);
+  }
+
+  getButtonText(param: string): string {
+    switch (param) {
+      case 'prepaid':
+        return 'Pay in Advance';
+      case 'postpaid':
+        return 'Pay what you used';
+      case 'monthly':
+        return 'Monthly rate';
+      case 'trial':
+        return 'Free trial';
+      case 'free':
+        return 'Free licence';
+      default:
+        return '';
+    }
+  }
+
+  ngOnInit(): void {
+    this.oneProductService
+      .getSingleProductDetails(this.route.snapshot.paramMap.get('id') || '')
+      .pipe(takeUntil(this.destroy$))
+      .subscribe({
+        next: (data) => {
+          this.oneProduct = data;
+        },
+      });
```

Comment:
Please move this out to a separate function and call it from ngOnInit.

Like this you get the chance to use a descriptive function name. If the name says what the function does I don't need to read the code to understand what is happening.
Also testing the code becomes easier.

Reply:
created a new function called "getProductDetailsById" which is now called in the ngOnInit function.